tests/include/opaque_test_driver.h

```diff
@@ -0,0 +1,175 @@
+ * to create an object that is equivalent to the public key.
+ *
+ * \param[in]  attributes   The attributes for the key.
+ * \param[in]  in           The opaque key to export.
+ * \param[in]  in_length    The size of the opaque key.
+ * \param[out] out          Buffer where the exported key data is to be written.
+ * \param[in]  out_size     Size of the \p out data buffer in bytes.
+ * \param[out] out_length   On success, the number of bytes that make up the
+ *                          exported key data.
+ *
+ * \retval #PSA_SUCCESS
+ * \retval #PSA_ERROR_INVALID_ARGUMENT
+ *         The \p in key is not recognized as an opaque key.
+ * \retval #PSA_ERROR_BUFFER_TOO_SMALL
+ *         The size of the \p out data buffer is too small.
+ */
+psa_status_t opaque_test_driver_export_public_key(
+                                        const psa_key_attributes_t *attributes,
+                                        const uint8_t *in,
+                                        size_t in_length,
+                                        uint8_t *out,
+                                        size_t out_size,
+                                        size_t *out_length );
+
+/**
+ * \brief Generate an opaque key.
+ *
+ * \param[in]  attributes   The attributes for the new key.
+ * \param[out] key          Buffer where the key data is to be written.
+ * \param[in]  key_size     Size of the \p key data buffer in bytes.
+ * \param[out] key_length   On success, the number of bytes that make up the
+ *                          key data.
+ *
+ * \retval #PSA_SUCCESS
+ * \retval #PSA_ERROR_INVALID_ARGUMENT
+ *         Zero length key.
+ * \retval #PSA_ERROR_NOT_SUPPORTED
+ *         Key length or type not supported.
+ * \retval #PSA_ERROR_BUFFER_TOO_SMALL
+ *         The size of the \p key data buffer is too small.
+ */
+psa_status_t opaque_test_driver_generate_key(
+                                        const psa_key_attributes_t *attributes,
+                                        uint8_t *key,
+                                        size_t key_size,
+                                        size_t *key_length );
+
+/**
+ * \brief Import a key and create an opaque key.
+ *
+ * \param[in]  attributes   The attributes for the key.
+ * \param[in]  in           The key to import.
+ * \param[in]  in_length    The size of the key.
+ * \param[out] out          Buffer where the opaque key data is to be written.
+ * \param[in]  out_size     Size of the \p out data buffer in bytes.
+ * \param[out] out_length   On success, the number of bytes that make up the
+ *                          opaque key data.
+ *
+ * \retval #PSA_SUCCESS
+ * \retval #PSA_ERROR_INVALID_ARGUMENT
+ *         The \p in key length is not matching \p attributes.
+ * \retval #PSA_ERROR_NOT_SUPPORTED
+ *         Key length or type not supported.
+ * \retval #PSA_ERROR_BUFFER_TOO_SMALL
+ *         The size of the \p out data buffer is too small.
```

Comment:
A driver import entry point shouldn't return `BUFFER_TOO_SMALL`. If the context is smaller than expected, that indicates a bug in the core, and the driver should convey _that_.